pallets/token_migration/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+            let migrator = ensure_signed(origin)?;
+            Self::migrate_(migrator, recipients)
+        }
+
+        /// Increase the migrators allowed migrations by the given number
+        #[weight = (0, Pays::No)]
+        pub fn expand_migrator(origin, migrator: T::AccountId, increase_migrations_by: u16) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            match Self::migrators(&migrator) {
+                Some(current_migrations) => {
+                    let new_migrations = current_migrations.checked_add(increase_migrations_by).ok_or(Error::<T>::CannotExpandMigrator)?;
+                    Migrators::<T>::insert(migrator.clone(), new_migrations.clone());
+                    Self::deposit_event(RawEvent::MigratorExpanded(migrator, new_migrations));
+                    Ok(())
+                },
+                None => fail!(Error::<T>::UnknownMigrator)
+            }
+        }
+
+        /// Decrease the migrators allowed migrations by the given number
+        #[weight = (0, Pays::No)]
+        pub fn contract_migrator(origin, migrator: T::AccountId, decrease_migrations_by: u16) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            match Self::migrators(&migrator) {
+                Some(current_migrations) => {
+                    let new_migrations = current_migrations.checked_sub(decrease_migrations_by).ok_or(Error::<T>::CannotContractMigrator)?;
+                    Migrators::<T>::insert(migrator.clone(), new_migrations.clone());
+                    Self::deposit_event(RawEvent::MigratorContracted(migrator, new_migrations));
+                    Ok(())
+                },
+                None => fail!(Error::<T>::UnknownMigrator)
+            }
+        }
+
+        /// Add a new migrator
+        #[weight = (0, Pays::No)]
+        pub fn add_migrator(origin, migrator: T::AccountId, allowed_migrations: u16) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            ensure!(!Migrators::<T>::contains_key(&migrator), Error::<T>::MigratorAlreadyPresent);
+            Migrators::<T>::insert(migrator, allowed_migrations);
+            Ok(())
+        }
+
+        /// Remove an existing migrator
+        #[weight = (0, Pays::No)]
+        pub fn remove_migrator(origin, migrator: T::AccountId) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            ensure!(Migrators::<T>::contains_key(&migrator), Error::<T>::UnknownMigrator);
+            Migrators::<T>::remove(migrator);
+            Ok(())
+        }
+    }
+}
+
+impl<T: Trait> Module<T> {
+    /// Deduct tokens from the migrator's account
+    pub fn migrate_(
+        migrator: T::AccountId,
+        recipients: BTreeMap<T::AccountId, BalanceOf<T>>,
+    ) -> dispatch::DispatchResult {
+        let allowed_migrations = Self::migrators(&migrator).unwrap();
+        let mut mig_count = recipients.len() as u16;
```

Comment:
Might want to guard against large recipients list (> 65535) overflowing u16.

Reply:
The cast will guard by truncating > `u16::MAX` but I see how that's swallowing an error. It won't matter in practice as such a large extrinsic will exceed max length and weight allowed in a block and thus won't be included.